File: bench/benches/fib.lua
-- naive recursive fibonacci -- function-call and integer-arithmetic stress.
package.path = (arg[0]:match("(.*/)") or "./") .. "../lib/?.lua;" .. package.path
local bench = require("bench")
local function fib(n) if n < 2 then return n else return fib(n - 1) + fib(n - 2) end end
bench("fib", function() return fib(30) end)

File: bench/benches/fib.java
class Main {
    static long fib(int n) {
        return n < 2 ? n : fib(n - 1) + fib(n - 2);
    }

    public static void main(String[] a) {
        Bench.bench("fib", () -> fib(30));
    }
}
